engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/streaming/embedded/EmbeddedDeploymentManager.scala

```diff
@@ -52,34 +53,49 @@ object EmbeddedDeploymentManager extends LazyLogging {
+    val deployedScenarios = Await.result(processingTypeDeploymentService.getDeployedScenarios, retrieveDeployedScenariosTimeout)
+    deployedScenarios.map(data => deployScenario(data.processVersion, data.deploymentData, data.resolvedScenario)._2).toMap
+  }
 
   override def deploy(processVersion: ProcessVersion, deploymentData: DeploymentData, processDeploymentData: ProcessDeploymentData, savepointPath: Option[String]): Future[Option[ExternalDeploymentId]] = {
-    parseScenario(processDeploymentData).map { scenario =>
-      val jobData = JobData(scenario.metaData, processVersion, deploymentData)
+    parseScenario(processDeploymentData).map { parsedResolvedScenario =>
+      deployScenarioClosingOldIfNeeded(processVersion, deploymentData, parsedResolvedScenario)
+    }
+  }
 
-      interpreters.get(processVersion.processName).foreach { case ScenarioInterpretationData(_, processVersion, oldVersion) =>
-        oldVersion.close()
-        logger.debug(s"Closed already deployed scenario: $processVersion")
-      }
-      val interpreter = new KafkaTransactionalScenarioInterpreter(scenario, jobData, modelData, contextPreparer)
-      val result = interpreter.run()
-      result.onComplete {
-        case Failure(exception) => handleUnexpectedError(processVersion, exception)
-        case Success(_) => //closed without problems
-      }
+  private def deployScenarioClosingOldIfNeeded(processVersion: ProcessVersion, deploymentData: DeploymentData, parsedResolvedScenario: EspProcess): Option[ExternalDeploymentId] = {
+    interpreters.get(processVersion.processName).foreach { case ScenarioInterpretationData(_, processVersion, oldVersion) =>
+      oldVersion.close()
+      logger.debug(s"Closed already deployed scenario: $processVersion")
+    }
+    val (deploymentId: String, deploymentEntry: (ProcessName, ScenarioInterpretationData)) = deployScenario(processVersion, deploymentData, parsedResolvedScenario)
+    interpreters += deploymentEntry
+    Some(ExternalDeploymentId(deploymentId))
+  }
 
-      val deploymentId = UUID.randomUUID().toString
-      interpreters += (processVersion.processName -> ScenarioInterpretationData(deploymentId, processVersion, interpreter))
-      logger.debug(s"Deployed scenario $processVersion")
-      Some(ExternalDeploymentId(deploymentId))
+  private def deployScenario(processVersion: ProcessVersion, deploymentData: DeploymentData, parsedResolvedScenario: EspProcess) = {
+    val jobData = JobData(parsedResolvedScenario.metaData, processVersion, deploymentData)
+    val interpreter = new KafkaTransactionalScenarioInterpreter(parsedResolvedScenario, jobData, modelData, contextPreparer)
+    val result = interpreter.run()
+    result.onComplete {
+      case Failure(exception) => handleUnexpectedError(processVersion, exception)
+      case Success(_) => //closed without problems
     }
+
+    val deploymentId = UUID.randomUUID().toString
```

Comment:
I wondered a bit if we should reuse the previous deploymentId when we restart the designer? But maybe it's just new deployment then...